front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainScheduleExportCSV.ts

```diff
@@ -0,0 +1,148 @@
+import type { PositionSpeedTime, SpeedPosition, Train } from 'reducers/osrdsimulation/types';
+import * as d3 from 'd3';
+import { BaseOrEcoType } from './DriverTrainScheduleTypes';
+
+/**
+ * CSV Export of trainschedule
+ *
+ * Rows : position in km, speed in km/h, speed limit in km/h, time in s
+ *
+ */
+
+enum CSVKeys {
+  op = 'op',
+  ch = 'ch',
+  lineCode = 'lineCode',
+  trackName = 'trackName',
+  position = 'position',
+  speed = 'speed',
+  speedLimit = 'speedLimit',
+  time = 'time',
+}
+
+type CSVData = {
+  [key in keyof typeof CSVKeys]: string;
+};
+
+type PositionSpeedTimeOP = PositionSpeedTime & {
+  speedLimit?: string | number;
+  op?: '';
+  ch?: '';
+  lineCode?: '';
+  trackName?: '';
+};
+
+const pointToComma = (number: number) => number.toString().replace('.', ',');
+
+const interpolateValue = (
+  position: number,
+  speedList: PositionSpeedTime[],
+  value: 'speed' | 'time'
+): number => {
+  const bisector = d3.bisectLeft(
+    speedList.map((d: SpeedPosition) => d.position),
+    position
+  );
+  if (speedList[bisector - 1]) {
+    const totalDistance = speedList[bisector].position - speedList[bisector - 1].position;
+    const distance = position - speedList[bisector - 1].position;
+    const totalDifference = speedList[bisector][value] - speedList[bisector - 1][value];
+    return speedList[bisector - 1][value] + (totalDifference * distance) / totalDistance;
+  }
+  return speedList[bisector][value];
+};
+
+const getStepSpeedLimit = (position: number, speedLimitList: Train['vmax']) => {
+  const bisector = d3.bisectLeft(
+    speedLimitList.map((d: SpeedPosition) => d.position),
+    position
+  );
+  return speedLimitList[bisector].speed || 0;
+};
+
+// Add OPs inside speedsteps array, gather speedlimit with stop position, and sort the array along position before return
+const overloadWithOPs = (train: Train, baseOrEco: BaseOrEcoType): PositionSpeedTimeOP[] => {
+  const trainRegime = train[baseOrEco];
+  if (trainRegime) {
```

Comment:
Move this above outside the function (in driverTrainScheduleExportCSV) and give the regime as an argument directly
```suggestion
```

Reply:
Very complicated regarding types: it introduces some types and consts necessities that complicates code...
The two propositions have pros and cons, and no one is better than another... I prefer let this as this if you're ok ?
I repeat: this code won't stay in the project.

Reply:
Resolved by our suggestion, thx!